src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node head;
+    private Node tail;
+
+    private int size;
+
     @Override
     public void add(T value) {
+        Node newNode = new Node(value, null, null);
+        if (isEmpty()) {
+            head = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexIsValid(index);
+        if (index == 0) {
+            head = new Node(value, null, head);
+            head.next.prev = head;
+        } else {
+            Node newNode = new Node(value, getNodeByIndex(index).prev, getNodeByIndex(index));
+            newNode.prev.next = newNode;
+            newNode.next.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T el : list) {
+            add(el);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node nodeToSet = getNodeByIndex(index);
+        T old = nodeToSet.value;
+        nodeToSet.value = value;
+        return old;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node nodeToRemove = getNodeByIndex(index);
+        unlink(nodeToRemove);
+        return nodeToRemove.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node currentNode = head;
```

Comment:
Look through your code and parametrize all your nodes
```suggestion
        Node<T> currentNode = head;
```